astropy/io/fits/card.py

```diff
@@ -959,7 +1053,98 @@ def _format_comment(self):
         else:
             return f' / {self._comment}'
 
-    def _format_image(self):
+    @property
+    def _full_unit_regexp(self):
+        s = re.escape(self.unit_str_start)
+        l = re.escape(self.unit_str_delimiters[0])
+        r = re.escape(self.unit_str_delimiters[1])
+        e = re.escape(self.unit_str_end)
+        unit_str_position = self.unit_str_position.lower()
+        if unit_str_position in ['start', 'beginning']:
+            ms = '^'
+            me = ''
+        elif unit_str_position == 'end':
+            ms = ''
+            me = '$'
+        else:
+            raise ValueError(f'Unknown unit string position {self.unit_str_position}.  Expecting "start" or "end"')
+        return f'{ms}{s}{l}.*{r}{e}{me}'
+
+    def _get_comment_unit(self, comment):
+        """Returns `~astropy.units.Unit` found in comment, if defined, else ``None``"""
+        # Extract unit together with lead-in and delimiter
+        # https://stackoverflow.com/questions/8569201/get-the-string-within-brackets-in-python
+        m = re.search(self._full_unit_regexp, comment)
+        if m is None:
+            #log.debug(f'no unit matching "{self._full_unit_regexp}" in "{comment}"')
+            return None
+        # Strip off delemeters
+        punit_str = m.group(0)
+        s = re.escape(self.unit_str_start)
+        l = re.escape(self.unit_str_delimiters[0])
+        r = re.escape(self.unit_str_delimiters[1])
+        e = re.escape(self.unit_str_end)
+        unit_str = re.sub(f'[{s}{l}{r}{e}]', '', punit_str)
+        try:
+            unit = u.Unit(unit_str)
```

Comment:
Possibly, we should parse the unit with `format='fits'`, to enforce consistency. On the other hand, on *reading*, it pays to be generous...

Actually, what we probably should do, is to add `parse_strict='silent'`, which means that an `UnrecognizedUnit` gets returned, which perhaps is more proper than ignoring it. Then we can leave out the `try/except`

Reply:
PERFECT!  The parse_strict='silent' ends up being the key!  That will enable parenthetical statements that are not intended to be units peacefully coexist with unit strings: e.g. keyword description (parenthetical) (s)